lib/view_component/base.rb

```diff
@@ -118,11 +118,11 @@ def with(area, content = nil, &block)
         raise ArgumentError.new "Unknown content_area '#{area}' - expected one of '#{content_areas}'"
       end
 
-      if block_given?
-        content = view_context.capture(&block)
+      define_singleton_method(area) do |*args|
+        block = ->() { content } unless block_given?
+        view_context.capture(*args, &block)
```

Comment:
This is the key change. Rather than capturing the block immediately and storing the result in the attr_reader instance variable we redefine the method and capture the block when its called passing the arguments.